plugins/module_utils/foreman_helper.py

```diff
@@ -680,6 +697,15 @@ def ensure_entity(self, resource, desired_entity, current_entity, params=None, s
 
         return updated_entity
 
+    def _validate_supported_payload(self, resource, action, payload):
+        filtered_payload = self._resource_prepare_params(resource, action, payload)
+        # On Python 2 dict.keys() is just a list, but we need a set here.
```

Comment:
On py3 it's an iterator afaik.
I see the reason for using sets nonetheless.

Reply:
```
>>> type({}.keys())
<class 'dict_keys'>
```

Whatever that is, but you can substract them as it'd be a `set`.

Reply:
right. That is a special view class. I thought it would react like an iterator, but maybe they made it subtractable.
We can leave the comment.